@xen-orchestra/lite/src/components/ActionButton.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import LoadingButton, { LoadingButtonProps } from '@mui/lab/LoadingButton'
+import { withState } from 'reaclette'
+
+interface ParentState {}
+
+interface State {
+  isLoading: boolean
+}
+
+interface Props extends LoadingButtonProps {
+  onClick: (e: React.MouseEvent) => Promise<void>
+}
+
+interface ParentEffects {}
+
+interface Effects {
+  _onClick: React.MouseEventHandler<HTMLButtonElement>
```

Comment:
if we have to perform an action with different parameters, how do we do it? for example: start a VM.

Reply:
My understanding is that you will declare the effect in the parent container, and transmit it to the `ActionButton` by the props `onClick`

Reply:
Yes.
But you can directly pass a fonction to the `onClick` prop and passing the params needed using `data-` prop.
eg:
`onClick={startVm}`
`data-vmId`={vmId}